### PR TITLE
Return the desktop mode as a fullscreen mode if no exclusive fullscre…

### LINKED ISSUE #101: video_getNumDisplayModes test fails under dummy video driver
To reproduce:

* Debian 12 in a qemu/kvm virtual machine (running GNOME 43 in Wayland mode, if that matters)
* SDL 3 libsdl-org/SDL@4db2b968 in `~/src/SDL3`, configured with `cmake -S . -B _build -DSDL_TESTS=ON` and compiled
* SDL 2 libsdl-org/SDL@9772d051 in `~/src/SDL2`, configured with `cmake -S . -B _build -DSDL_TESTS=ON` and compiled
* sdl2-compat 52d96367 in `~/src/sdl2-compat`, configured with `cmake -S . -B _build -DSDL3_INCLUDE_DIRS=$HOME/src/SDL3/include -DSDL2COMPAT_TESTS=ON` and compiled
* `export SDL_VIDEODRIVER=dummy SDL_AUDIODRIVER=dummy SDL_VIDEO_DRIVER=dummy SDL_AUDIO_DRIVER=dummy`
* `cd ~/src/SDL3/_build/test; ./testautomation` (all succeeds)
* `cd ~/src/SDL2/_build/test; ./testautomation` (there are some test failures which are out-of-scope for this bug report, but `video_getNumDisplayModes` succeeds)
* `cd ~/src/sdl2-compat/_build; LD_LIBRARY_PATH=$HOME/src/SDL3/_build ./testautomation`
* `cd ~/src/sdl2-compat/_build; LD_LIBRARY_PATH=$HOME/src/SDL3/_build ./testautomation --filter video_getNumDisplayModes`

Expected result: sdl2-compat test suite passes

Actual result:

```
INFO:  09/28/23 11:08:57: ----- Test Case 21.6: 'video_getNumDisplayModes' started
INFO:  09/28/23 11:08:57: Test Description: 'Use SDL_GetNumDisplayModes function to get number of display modes'
INFO:  09/28/23 11:08:57: Test Iteration 1: execKey 13259322882708337379
INFO:  09/28/23 11:08:57: Assert 'Call to SDL_GetNumVideoDisplays()': Passed
INFO:  09/28/23 11:08:57: Assert 'Call to SDL_GetNumDisplayModes(0)': Passed
ERROR: 09/28/23 11:08:57: Assert 'Validate returned value from function; expected: >=1; got: 0': Failed
ERROR: 09/28/23 11:08:57: Assert Summary: Total=3 Passed=2 Failed=1
INFO:  09/28/23 11:08:57: Total Test runtime: 0.0 sec
ERROR: 09/28/23 11:08:57: >>> Test 'video_getNumDisplayModes': Failed
```

### PR DESCRIPTION
**Pull request by Kontrabant**
…en modes are available

It's perfectly valid for a video driver in SDL3 to not expose any exclusive fullscreen modes, however, SDL2 tests and applications generally expect that at least one mode will be exposed, typically one that matches the desktop mode if nothing else. If no exclusive fullscreen modes are exposed by the SDL3 video driver, the sdl2-compat layer will expose the desktop mode as a fullscreen mode. A discrepancy between the real SDL2 and the compatibility layer is also addressed, as in SDL2, the desktop fullscreen flag is always combined with the general fullscreen flag, which wasn't the case in sdl2-compat.

This more closely mimics the behavior of the real SDL2 and results in all of the related tests passing.

Fixes #101

### CODE REVIEW

**Comment by sezero** [comment]
The following fixes the build warnings and errors:
```diff
diff --git a/src/sdl2_compat.c b/src/sdl2_compat.c
index f61d2f8..4693d15 100644
--- a/src/sdl2_compat.c
+++ b/src/sdl2_compat.c
@@ -716,3 +716,3 @@ BOOL WINAPI _DllMainCRTStartup(HANDLE dllhandle, DWORD reason, LPVOID reserved)
 #define SDL2_WINDOW_SHOWN 0x000000004
-#define SDL2_WINDOW_FULLSCREEN_DESKTOP 0x00001000 | SDL_WINDOW_FULLSCREEN
+#define SDL2_WINDOW_FULLSCREEN_DESKTOP (0x00001000 | SDL_WINDOW_FULLSCREEN)
 #define SDL2_WINDOW_SKIP_TASKBAR 0x00010000
@@ -4548,4 +4548,4 @@ SDL_GetDisplayModeList(SDL_DisplayID displayID, int *count)
 
-    SDL3_free(modes);
-    modes = SDL3_calloc(sizeof(SDL_DisplayMode*), 2);
+    SDL3_free((void *)modes);
+    modes = (const SDL_DisplayMode **) SDL3_calloc(sizeof(SDL_DisplayMode*), 2);
     if (modes) {
```

**Review by sezero — APPROVED** [review]

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -4529,6 +4529,37 @@ Display_IDToIndex(SDL_DisplayID displayID)
     return displayIndex;
 }
 
+static const SDL_DisplayMode**
+SDL_GetDisplayModeList(SDL_DisplayID displayID, int *count)
+{
+    int c;
```
Comment (slouken) [inline src/sdl2_compat.c:4535]: This should probably be initialized to 0

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -4529,6 +4529,37 @@ Display_IDToIndex(SDL_DisplayID displayID)
     return displayIndex;
 }
 
+static const SDL_DisplayMode**
+SDL_GetDisplayModeList(SDL_DisplayID displayID, int *count)
```
Comment (slouken) [inline src/sdl2_compat.c:4533]: In general we're moving these onto the same line, e.g.
```
static const SDL_DisplayMode **SDL_GetDisplayModeList(SDL_DisplayID displayID, int *count)
```

**Comment by sezero** [comment]
@slouken: Good to go now?

**Review by slouken — APPROVED** [review]

**Pull request merged**